pkg/scheduler/framework/plugins/noderesources/fit_test.go

```diff
@@ -715,7 +726,24 @@ func TestRestartableInitContainers(t *testing.T) {
 		{
 			name:                    "allow pod with restartable init containers if sidecar containers is enabled",
 			enableSidecarContainers: true,
-			pod:                     newPodWithRestartableInitContainers(),
+			pod:                     newPodWithRestartableInitContainers(nil, nil),
+		},
+		{
+			name:                    "When sidecar does not request resources, the pod passes the filter.",
+			enableSidecarContainers: true,
+			pod: newPodWithRestartableInitContainers(
+				&v1.ResourceList{v1.ResourceCPU: *resource.NewMilliQuantity(1, resource.DecimalSI)},
+				nil,
+			),
+		},
+		{
+			name:                    "When sidecar requests resources, the pod can't pass the filter.",
+			enableSidecarContainers: true,
+			pod: newPodWithRestartableInitContainers(
+				&v1.ResourceList{v1.ResourceCPU: *resource.NewMilliQuantity(1, resource.DecimalSI)},
+				&v1.ResourceList{v1.ResourceCPU: *resource.NewMilliQuantity(1, resource.DecimalSI)},
+			),
+			wantFilterStatus: framework.NewStatus(framework.Unschedulable, "Insufficient cpu"),
```

Comment:
Was this behavior buggy or are you just adding coverage? If it was buggy, then it needs to be expressed in the release note.

Also, the reason it doesn't pass the filter is because the sum of resources would just go over the node allocatable, not because there are sidecar requests.

We need another test that has sidecar requests but still passes the filter.

Reply:
> Was this behavior buggy or are you just adding coverage?

I was just adding coverage.